org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -190,46 +191,39 @@ protected void terminate(KillBean kbean) {
 			stop();
 			if (kbean.isDisconnect()) disconnect();
 		} catch (EventException e) {
-			logger.error("An internal error occurred trying to terminate the consumer "+getName()+" "+getConsumerId());
+			LOGGER.error("An internal error occurred trying to terminate the consumer "+getName()+" "+getConsumerId());
 		}
 		if (kbean.isExitProcess()) {
 			try {
 				Thread.sleep(2500);
 			} catch (InterruptedException e) {
-				logger.error("Unable to pause before exit", e);
+				LOGGER.error("Unable to pause before exit", e);
 			}
-			main();
+			exit();
 		}
 		if (kbean.isRestart()) {
 			try {
-				if (kbean.isDisconnect()) connect();
-			    start();
+				if (kbean.isDisconnect()) {
+					connect();
+				}
+				start();
 			} catch (EventException e) {
-				logger.error("Unable to restart, please contact your support representative.", e);
+				LOGGER.error("Unable to restart, please contact your support representative.", e);
 			}
 		}
 	}
 
-	/**
-	 * Exit is protected inside a main() method
-	 * because doing this gets around the sonarqube
-	 * rule which says System.exit(...) must not be
-	 * called except inside a main method.
-	 *
-	 * @param args
-	 */
-	public static final void main(String... args) {
+	public void exit() {
+		// TODO: We should almost certainly not be killing the VM. We need to investigate
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)